src/harness/testcases/WINNF_FT_S_REG_testcase.py

```diff
@@ -0,0 +1,262 @@
+    self._sas_admin.InjectUserId({'userId': device_b['userId']})
+    self._sas_admin.InjectUserId({'userId': device_c['userId']})
+    self._sas_admin.InjectUserId({'userId': device_d['userId']})
+
+    self._sas_admin.PreloadRegistrationData(conditionals)
+
+    # Device 2 missing cbsdSerialNumber
+    del device_b['cbsdSerialNumber']
+
+    # Device 3 missing fccId
+    del device_c['fccId']
+
+    # Device 4 missing userId
+    del device_d['userId']
+
+    # Register devices
+    devices = [device_a, device_b, device_c, device_d]
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)
+    # Check registration response
+    self.assertTrue('cbsdId' in response['registrationResponse'][0])
+    self.assertEqual(
+        response['registrationResponse'][0]['response']['responseCode'], 0)
+    for x in range(1, 4):
+      self.assertEqual(
+          response['registrationResponse'][x]['response']['responseCode'], 102)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_REG_9(self):
+    """Blacklisted CBSD in Array Registration request (responseCode 101).
+
+    The response should be BLACKLISTED 101.
+    """
+
+    # Load Devices
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    devices = [device_a, device_b, device_c]
+    for device in devices:
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    for resp in response:
+      self.assertTrue('cbsdId' in resp)
+      self.assertEqual(resp['response']['responseCode'], 0)
+    del request, response
+
+    # Blacklist the third device
+    self._sas_admin.BlacklistByFccId({'fccId': device_c['fccId']})
+
+    # Re-register the devices
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)['registrationResponse']
+
+    # Check registration response
+    self.assertEqual(len(response), len(devices))
+    for resp in response[:2]:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      self.assertTrue('cbsdId' in resp)
+    self.assertEqual(response[2]['response']['responseCode'], 101)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_REG_10(self):
+    """Unsupported SAS protocol version in Array request (responseCode 100).
+
+    The response should be FAILURE 100.
+    """
+
+    # Use higher than supported version
+    self._sas._sas_version = 'v2.0'
+
+    # Load Devices
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_b = json.load(
```

Comment:
same comment as in test Id #9, we should use all cat A CBSDs for this test

Reply:
Done.